classes/helper.php

```diff
@@ -341,13 +341,16 @@ public static function retrieve_user_notices(): array {
                 continue;
             }
             $notice = $notices[$noticeid];
+            $dissmised = $data['action'] == acknowledgement::ACTION_DISMISSED;
```

Comment:
`dissmised` -> `dismissed`

And place use `===` instead of `==`, as it should be an exact match.

Reply:
that actually the fix for the issue. In some cases it comes as 0 and in some as "0". I think it's ok to have == here. IN this case we don't really care if it' 0 "0" or false. All the same.

Reply:
Ah that's the reason. I think left in its current state, it might be a bit dangerous for a future contributor who might go through and clean up the code. 

As the acknowledgement states seem to only be numbers (e.g. int), it might have been better to cast the action to an int, and use it in the comparison, rather than having a looser check. In this case, having the more explicit check/cast would inform the person making the changes that it's there for a reason, as opposed to a loose check which they might feel like adding an extra `=` (as I suggested). Alternatively, a comment for the looser check might have been helpful here.

Reply:
Agree. The code is not in the best shape. There is open issue already https://github.com/catalyst/moodle-local_sitenotice/issues/26 Can you please add you thoughts there please. I hope one day we will clean it up...